src/components/EditorSideBar/index.tsx

```diff
@@ -1,15 +1,40 @@
 import EditorSideBarItem from "./EditorSideBarItem";
 import sideBarItemList from "./const";
 import { Box, Flex } from "@chakra-ui/react";
+import EditorSideBarContent from "./EditorSideBarContent";
+import { useState } from "react";
 
 const EditorSideBar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [iconClick, setIconClick] = useState("");
+
+  const handleIconClick = (title: string) => {
+    setIsOpen(true);
+    setIconClick(title);
+  };
+  const handleCloseClick = () => {
+    setIsOpen(false);
+    setIconClick("");
+  };
+
   return (
-    <Box bg="background.primary" w="98px" h="100vh" position="fixed" top="72px" left="0">
-      <Flex align="center" justify="center" flexDirection="column" maxW="98px">
-        {sideBarItemList.map((item) => (
-          <EditorSideBarItem key={item.title} title={item.title} icon={item.icon} />
-        ))}
-      </Flex>
+    <Box>
+      <Box bg="background.primary" w="98px" h="100vh" position="fixed" top="72px" left="0">
+        <Flex align="center" justify="center" flexDirection="column" maxW="98px">
+          {sideBarItemList.map((item) => (
+            <EditorSideBarItem
+              key={item.title}
+              title={item.title}
+              icon={item.icon}
+              onHandleIconClick={() => handleIconClick(item.title)}
+              iconClick={iconClick}
+            />
+          ))}
+        </Flex>
+      </Box>
+      {isOpen && (
+        <EditorSideBarContent iconClick={iconClick} onHandleCloseClick={handleCloseClick} />
+      )}
```

Comment:
You can use some of the chakra built-in transition components to make this popup window animate smooth.
![image](https://user-images.githubusercontent.com/12379895/169649030-9dff8eaa-a6e5-43cd-a45b-1a74b0cb3af6.png)

Reply:
Talked with BA and might design it later.